toptal/picasso
# [CPT-1478] Add enableResetSearch prop to Select and QB

[CPT-1478](https://toptal-core.atlassian.net/browse/CPT-1478)

### Description

In this PR we provided new prop enableResetChange for Select component in order to be able to reset value from the search input and this prop was added to Multiselect component for QB
### How to test


- [Temploy](https://picasso.toptal.net/cpt-1478-add-clear-feature-to-select-search-in-picasso)
- Open temploy
- Go to Select story
- Under `Search behavior` example check if new example `Search with input reset` appeared
- Make sure that reset is working as expected (when something is typed clicking on reset button clears the value)

### Screenshots

| Before.                                 | After.                                  |
| --------------------------------------- | --------------------------------------- |
| ![image](https://github.com/toptal/picasso/assets/112090359/2df26d72-b58e-4bce-87f5-e7e768a5bb82) |![image](https://github.com/toptal/picasso/assets/112090359/b57c7786-765e-4271-bdb2-2f170bec53ab)|

### Development checks


**Breaking change**

- n/a codemod is created and showcased in the changeset

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>


[CPT-1478]: https://toptal-core.atlassian.net/browse/CPT-1478?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
